variables.tf

```diff
@@ -689,6 +689,13 @@ variable "maintenance_window_node_os" {
 EOT
 }
 
+variable "msi_auth_for_monitoring_enabled" {
+  type        = bool
+  default     = false
```

Comment:
Why you have the default at false and not null ?

https://github.com/hashicorp/terraform-provider-azurerm/blob/8078e1bcdb8503e6c3a581f91aa3b201c4545f5c/internal/services/containers/kubernetes_addons.go#L121

Reply:
It is weird, because enabling monitoring addon with `az aks` CLI the default value is True..

Reply:
Also is nullable = false, I interpreted it as only possible values should be either False or True? Correct me if I'm wrong

Reply:
@admincasper It is often the case that Terraform has `true` `false` and `null`. But when you read back from the azurerm API the values are just `true` or `false`. Now we are interested if this value is explicit in the Terraform state as `false`. I hope this clarifies why you dont get back `null` from your `az aks show` operation.

Reply:
Yeah thanks for clarifying! I've pushed up the changes.